svg.js.d.ts

```diff
@@ -548,13 +548,15 @@ declare module "@svgdotjs/svg.js" {
         to(a: any): Morphable;
     }
 
+    type ListEachCallback<T> = (el: T, index: number, list: T[]) => T
+
     // List.js
-    interface List<T> extends BuiltInArray<T> {
+    class List<T> extends BuiltInArray<T> {
         // I have no clue how to deal with this
         // [key: string]: (...arg0: any[]) => List<T>
         // [key: string]: () => List<any>
-        each(fn: Function): void
-        each(...args: any[]): void
+        each(fn: ListEachCallback<T>): T[]

```

Comment:
Actually this  method can take a function or a string and it doesnt return always an array of T. Whatever the function return you pass to it (as function or string name) will be maped into the returned array.
This is pretty much impossible to type -.-. But it always returns a List!

```suggestion
        each(fn: ListEachCallback<T>): List<any>
```

e.g. `list.each('fill')` will return a new list with contains all the fill colors of the elements in the list because it acts as getter. Or `list.each('animate')` will contain lots of runners.

However, the string version actually needs to go onto the next line...

Reply:
I don't think returns a List: isn't it just a lower-case array?

How about this:

~~~ js
    class List<T> extends BuiltInArray<T> implements ElementAlias {
        each(fn: ListEachCallback<T>): any[]
        each(name: string, ...args: any[]): any[]
        toArray(): T[]
    }
~~~

Does implementing `ElementAlias` cover the interface? Would we need Runner as well?

Reply:
No, it does indeed return a new list. The code is `this.map(fn)` which will create a new List. You can try it for yourself.

ElementAlias is not an interface. It is just all svg.js types as a union. You would need an interface with ALL the methods which are available on any element of svg.js.